### PR TITLE
Step Selector: Disable selector and table when Horizontal Axis is not set to Step.

### PR DESCRIPTION
**Pull request by JamesHollyer**
* Motivation for features / changes
We have not implemented all that is needed to make this feature work for other types of axis. Instead of implementing this in Phase 1 we are just disabling this when the axis is not set to Step.

* Technical description of changes
I basically just utilized the getMetricsXAxisType selector to disable and enable the proper pieces. This started to make the calculation a bit complex to have directly in the html file so I often moved it to a utility function in the component.  I also utilized to "title" attribute to give a tooltip which hopefully will alleviate any confusion users may have.

* Screenshots of UI changes
Here is what the settings looks like when disabled and the mouse if hovering over the step selector setting.
Screen Shot 2022-07-28 at 5 31 20 PM

* Alternate designs / implementations considered
I considered just implementing the pieces to enable relative time as well. The problem is implementing a way to allow typing the time.

### CODE REVIEW

**Review by bmd3k — APPROVED** [review]

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_component.ts [hunk 1]
```diff
@@ -318,4 +318,12 @@ export class ScalarCardComponent<Downloader> {
       );
     }
   }
+
+  isInStepSelectorMode(): boolean {
```
[the remark this answers is not in the document]
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_component.ts]: Changed to inTimeSelectionMode.

File: tensorboard/webapp/metrics/views/main_view/card_grid_component.ts [hunk 2]
```diff
@@ -47,6 +48,7 @@ export class CardGridComponent {
   @Input() cardObserver!: CardObserver;
   @Input() showPaginationControls!: boolean;
   @Input() isStepSelectorEnabled!: boolean;
+  @Input() getMetricsXAxisType!: XAxisType;
```
[the remark this answers is not in the document]
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/main_view/card_grid_component.ts]: Done.

File: tensorboard/webapp/metrics/views/right_pane/settings_view_component.ng.html [hunk 3]
```diff
@@ -59,6 +59,8 @@ <h3 class="section-title">General</h3>
     <mat-checkbox
       [checked]="isScalarStepSelectorEnabled"
       (change)="stepSelectorToggled.emit()"
+      [disabled]="!isAxisTypeStep()"
+      title="Only available When Horizontal Axis is set to step"
```
[the remark this answers is not in the document]
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/right_pane/settings_view_component.ng.html]: Yea I guess you are right. Changed.

**Review by JamesHollyer — COMMENTED** [review]
Thanks Brian.

**Pull request merged**